core/variable_operations.cpp

```diff
@@ -294,4 +294,52 @@ Variable masks_merge_if_contained(const MasksConstProxy &masks,
   return mask_union;
 }
 
+VariableProxy replace_nan(const VariableConstProxy &var, double replacement,
+                          const VariableProxy &out) {
+
+  if (var.dtype() != out.dtype())
+    throw except::TypeError("Input and output variable types do not match");
+  if (var.dtype() != dtype<decltype(replacement)>)
+    throw except::TypeError("Replacement type doesn't match type of input");
+
+  transform_in_place<pair_self_t<double, float>>(
+      out, var,
+      scipp::overloaded{
+          [&](auto &x, const auto &y) {
+            if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(x)>> &&
+                          is_ValueAndVariance_v<std::decay_t<decltype(y)>>) {
```

Comment:
Instead of handling values and variances in this way it is sometimes easier to provide the required functions for `ValueAndVariance`, i.e., define `isnan(const ValueAndVariance &)`, use `using std::isnan`, and rely on ADL.

Reply:
Ok. I guess I can put that with the other operators alongside ValueAndVariance.